src/tryCatch.js

```diff
@@ -0,0 +1,34 @@
+var _arity = require('./internal/_arity');
+var _concat = require('./internal/_concat');
+var _curry2 = require('./internal/_curry2');
+
+
+/**
+ * `tryCatch` takes two functions, a `tryer` and a `catcher`. The returned
+ * function evaluates the `tryer`; if it does not throw, it simply returns the
+ * result. If the `tryer` *does* throw, the returned function evaluates the
+ * `catcher` function and returns its result. Note that for effective
+ * composition with this function, both the `tryer` and `catcher` functions
+ * must return the same type of results.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (...x -> a) -> ((e, ...x) -> a) -> (...x -> a)
+ * @param {Function} tryer The function that may throw.
+ * @param {Function} catcher The function that will be evaluated if `tryer` throws.
+ * @return {Function} A new function that will catch exceptions and send then to the catcher.
+ * @example
+ *
+ *      R.tryCatch(R.prop('x'), R.F, {x: true}); //=> true
+ *      R.tryCatch(R.prop('x'), R.F, null);      //=> false
+ */
+module.exports = _curry2(function _tryCatch(tryer, catcher) {
+  return _arity(tryer.length, function() {
+    try {
+      return tryer.apply(this, arguments);
+    } catch (e) {
+      return catcher.apply(this, _concat([e], arguments));
```

Comment:
i went back and forth on where to put the exception. I opted for first because 

1. the fact that something exceptional has happened is the most important thing. the original arguments are somewhat of a "by the way" concern;
2. error-first means you can re-use the same `catcher` in a generic way, i.e. without regard to the number of arguments.